spec/cc/workspace/path_tree_benchmark_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "spec_helper"
+
+module CC::Workspace
+  describe PathTree do
+    include FileSystemHelpers
+
+    MAX_FILTER_SECONDS = 10
+    NUM_FILES = 10_000
+    AVG_FILES_PER_DIR = 20
+    REPO_DEPTH = 30
+    NUM_EXCLUDES = 50
+
+    it "performs acceptably" do
```

Comment:
This spec may be better suited to a different `rake` task outside of `test` -- it's not a traditional spec in many ways. But since we got into our previous bad situation via a number of well-intentioned steps that led us to an unfortunate position, I wanted some kind of automated check to prevent this performance getting out of hand again.